Tests/MapboxNavigationTests/MapViewTests.swift

```diff
@@ -137,4 +138,151 @@ class MapViewTests: XCTestCase {
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in English after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "es"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_es" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in Spanish after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "zh-Hant-TW"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_zh-Hant" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in Traditional Chinese after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "zh-Hans-CN"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        // https://github.com/mapbox/mapbox-maps-ios/issues/652
+        XCTAssertNotEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                          Exp(.format) {
+                            Exp(.coalesce) { Exp(.get) { "name_zh-Hans" }; Exp(.get) { "name" } }
```

Comment:
Localization is currently broken for most Chinese locales, including Simplified Chinese and Hong Kong Traditional Chinese: mapbox/mapbox-maps-ios#652.

Reply:
I worked around this issue for now by ensuring that only supported locales make it through to the map SDK.